python/ray/dag/compiled_dag_node.py

```diff
@@ -907,11 +895,31 @@ def run(self):
         monitor.start()
         return monitor
 
+    def _execute_until(
+        self,
+        execution_index: int,
+    ) -> Any:
+        """Repeatedly execute this DAG until the given execution index.
+
+        Args:
+            execution_index: The execution index to execute until.
+
+        Returns:
+            The execution result corresponding to the given execution index.
+        """
+        while execution_index >= len(self._result_buffer):
+            # TODO: limit _result_buffer size
```

Comment:
I'm assuming this will be included in the PR?

Reply:
Yes. Done.